src/vs/workbench/contrib/outline/browser/outlinePane.ts

```diff
@@ -355,13 +371,14 @@ export class OutlinePane extends ViewPane {
 			}
 		}));
 
-		// last: set tree property
+		// last: set tree property and wire it up to one of our context keys
 		tree.layout(this._treeDimensions?.height, this._treeDimensions?.width);
 		this._tree = tree;
 		this._editorControlDisposables.add(toDisposable(() => {
 			tree.dispose();
 			this._tree = undefined;
 		}));
+		this._disposables.add(this._tree.onDidChangeCollapseState(() => this._updateAllCollapsedContext()));
```

Comment:
This will leak because listeners are piling up in the object disposable store, not the session store...

Reply:
Thanks for fixing this.